apps/kg/dataloader/sampler.py

```diff
@@ -8,15 +8,29 @@
 import pickle
 import time
 
-# This partitions a list of edges based on relations to make sure
-# each partition has roughly the same number of edges and relations.
-def RelationPartition(edges, n):
+# This partitions a list of edges based on relations and make sure
+# each relation only fall into one partition.
+def StrictRelationPartition(edges, n):
     heads, rels, tails = edges
-    print('relation partition {} edges into {} parts'.format(len(heads), n))
     uniq, cnts = np.unique(rels, return_counts=True)
     idx = np.flip(np.argsort(cnts))
     cnts = cnts[idx]
     uniq = uniq[idx]
+    for i in range(10):
+        print("ID-{}/{}:{}".format(uniq[i], len(cnts), cnts[i]))
+
+    threashold = 10000000
+    num = 0
+    for i in range(len(uniq)):
+        cnt = cnts[i]
+        if cnt >= threashold:
+            num += 1
+        else:
+            print(">{}:{}".format(threashold, num))
+            threashold //= 10
+            num = 0
+    print(">{}:{}".format(threashold, num))
```

Comment:
what is this for?

Reply:
For relation info statistics.
I will remove it latter